src/main/java/aspects/SetBrowserValuesAspect.java

```diff
@@ -0,0 +1,32 @@
+package aspects;
+
+import beans.AuthenticatedRequestBean;
+import beans.SessionNavigationBean;
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.javarosa.core.model.utils.DateUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.annotation.Order;
+import services.BrowserValuesProvider;
+
+/**
+ * Created by amstone326 on 1/8/18.
+ */
+@Aspect
+@Order(1)
+public class SetBrowserValuesAspect {
+
+    @Autowired
+    private BrowserValuesProvider browserValuesProvider;
+
+    @Before(value = "@annotation(org.springframework.web.bind.annotation.RequestMapping)")
+    public void setValues(JoinPoint joinPoint) throws Throwable {
+        Object beanArg = joinPoint.getArgs()[0];
+        if (beanArg instanceof AuthenticatedRequestBean) {
+            browserValuesProvider.setTimezoneOffset((AuthenticatedRequestBean)beanArg);
+        }
+        DateUtils.setTimezoneProvider(browserValuesProvider);
```

Comment:
What happens if there isn't a value for `tz_offset_millis`?

Reply:
The default value of `timezoneOffsetMillis` in `AuthenticatedRequestBean` is also -1, so if the request body never sets a different value, then the call to `bean.getTzOffset()` in `BrowserValuesProvider.setTimezoneOffset()` will return -1, thus resulting in the same situation I described in the above question.